gke-disk-image-builder/script/startup.sh

```diff
@@ -84,14 +84,32 @@ function process_snapshots() {
   for snapshot in "${snapshots[@]}"; do
     echo Processing $snapshot
 
-    sudo ctr -n k8s.io snapshot view tmp_$snapshot $snapshot
-    # Check if the view was successfully created
-    if [ $? -ne 0 ]; then
-      echo Failed to create snapshot view for $snapshot. Please rerun the tool to try it again.
+    # Add retry in case `ctr snapshot view` fails silently
+    local retries=5
+    while [ ${retries} -ge 0 ]; do 
+      sudo ctr -n k8s.io snapshot view tmp_$snapshot $snapshot
+      # Check if the view was successfully created
+      if [ $? -ne 0 ]; then
+        echo Failed to create snapshot view for $snapshot. Will retry.
+        continue
+      fi
+      
+      original_path=$(sudo ctr -n k8s.io snapshot mount /tmp_$snapshot tmp_$snapshot | grep -oP '/\S+/snapshots/[0-9]+/fs' | tr ':' '\n' | head -n 1)
+      if [[ -n "$original_path" ]]; then
+        break
+      fi
+      echo "Failed to get mount point for tmp_$snapshot. Will retry. ${retries} retries left."
+      echo "All snapshots:"
+      sudo ctr -n k8s.io snapshot list
+      sudo ctr -n k8s.io snapshot rm tmp_$snapshot
+      ((retries--))
```

Comment:
Maybe move `((retries--))` to the beginning of the loop. The `continue` branch ("Failed create snapshot view") does not change loop counter $retries, and that could cause endless loop I guess.

Reply:
uh good catch. Done